tests/lib/test_archive.py

```diff
@@ -169,6 +170,12 @@ def test_download(tmp_path: pathlib.Path, mock_sleep: mock.MagicMock) -> None:
     )
 
     dest = f"{tmp_path}/a"
+
+    # just exercising the delete symlink if exists path here than copypaste this entire test
+    with open(f"{tmp_path}/hi", "wb"):
+        pass
```

Comment:
what does it do for a dead symlink

Reply:
a dead symlink'll still be removed since os.path.islink is still True

(but os.path.exists will be False which is why it goes after)

Reply:
more specifically this should be a test too I think

Reply:
done!